src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1475 @@
+{
+    private int [|_goo|];
+    public int M1() => M2(_goo);
+    public int M2(in int i) => { i = 0; return i; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsRefArg()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public int M1() => M2(ref _goo);
+    public int M2(ref int i) => i;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOutArg()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public int M1() => M2(out _goo);
+    public int M2(out int i) => { i = 0; return i; }
+}",
+@"class MyClass
+{
+    public int M1() => M2(out _goo);
+    public int M2(out int i) => { i = 0; return i; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task MethodIsArg()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|M|]() => 0;
+    public int M1() => M2(M);
+    public int M2(System.Func<int> m) => m();
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsArg()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] => 0;
+    public int M1() => M2(P);
+    public int M2(int p) => p;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsArg()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+    public int M1(int x) => M2(this[x]);
+    public int M2(int p) => p;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsArg()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|_e|];
+    public EventHandler M1() => M2(_e);
+    public EventHandler M2(EventHandler e) => e;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task MultipleFields_AllUnused()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|] = 0, _bar = 0;
+}",
+@"class MyClass
+{
+    private int _bar = 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task MultipleFields_AllUnused_02()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int _goo = 0, [|_bar|];
+}",
+@"class MyClass
+{
+    private int _goo = 0;
```

Comment:
do we have fix-all tests for this situation?

Reply:
Yes, FixAll tests are at the end.

---
In reply to: [212812629](https://github.com/dotnet/roslyn/pull/29511#discussion_r212812629) [](ancestors = 212812629)